src/Fixer/ArrayNotation/YieldFromArrayToYieldsFixer.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\ArrayNotation;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Kuba Werłos <werlos@gmail.com>
+ */
+final class YieldFromArrayToYieldsFixer extends AbstractFixer
+{
+    /**
+     * @var array<int, Token>
+     */
+    private array $inserts;
```

Comment:
myself, I am trying to keep Fixers stateless, ie not having interim states in properties (ideally all properties are readonly and injected via constructor (yeah, we don't have DIC yet)).

Here, we populate this property in one method, consume in another, JohnDoe need to be mindful to not break interim state of this property and call functions in proper order.

I would suggest to change it - https://github.com/6b7562617765726c6f73/PHP-CS-Fixer/pull/2/commits/a49645015b1a8fb217b7a049cc5da0a5fc6fef9f